security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -189,6 +189,15 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, proxyID, resourceName
 
 // SecretExist checks if secret already existed.
 func (sc *SecretCache) SecretExist(proxyID, resourceName, token, version string) bool {
+	// If node agent works as ingress gateway agent, searches for kubernetes secret.
```

Comment:
could you explain a little bit why this check is needed ? ingress secret is also stored ```sc.secrets``` and existing logic should cover this ?